web/packages/shared/components/TerminalSearch/TerminalSearch.tsx

```diff
@@ -0,0 +1,184 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { useEffect, useState, useRef, useCallback } from 'react';
+import styled, { useTheme } from 'styled-components';
+import { Flex, ButtonIcon, Box, P2, Input } from 'design';
+import { Terminal as Xterm } from '@xterm/xterm';
+import { ISearchOptions } from '@xterm/addon-search';
+import { ArrowDown, ArrowUp, Cross } from 'design/Icon';
+
+export const TerminalSearch = ({
+  xTerm,
+  show,
+  onClose,
+  onOpen,
+}: {
+  xTerm: TerminalWithSearch;
+  show: boolean;
+  onClose(): void;
+  onOpen(): void;
+}) => {
+  const theme = useTheme();
+  const searchInputRef = useRef<HTMLInputElement>();
+  const [searchValue, setSearchValue] = useState('');
+  // const [showSearch, setShowSearch] = useState(false);
+  const [searchResults, setSearchResults] = useState<{
+    resultIndex: number;
+    resultCount: number;
+  }>({ resultIndex: 0, resultCount: 0 });
+
+  function search(value: string, prev?: boolean) {
+    if (!xTerm) {
+      return;
+    }
+    const match = theme.colors.terminal.brightYellow;
+    const activeMatch = theme.colors.terminal.yellow;
+    setSearchValue(value);
+
+    xTerm.search(value, prev, {
+      decorations: {
+        matchOverviewRuler: match,
+        activeMatchColorOverviewRuler: activeMatch,
+        matchBackground: match,
+        activeMatchBackground: activeMatch,
+      },
+    });
+  }
+
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    search(e.target.value);
+  };
+
+  const onKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter') {
+      search(e.currentTarget.value);
+    }
+  };
+
+  function searchNext() {
+    search(searchValue);
+  }
+
+  function searchPrevious() {
+    search(searchValue, true /* search for previous result */);
+  }
+
+  const onEscape = useCallback(() => {
+    onClose();
+    xTerm.clearSearch();
+    xTerm.term.focus();
+  }, [xTerm, onClose]);
+
+  const onSearchResultsChange = useCallback(
+    (resultIndex: number, resultCount: number) => {
+      setSearchResults({ resultIndex, resultCount });
+    },
+    []
+  );
+
+  useEffect(() => {
+    if (!xTerm) {
+      return;
+    }
+    xTerm.loadSearchAddon({ onDidChangeResults: onSearchResultsChange });
+    const listener = (e: KeyboardEvent) => {
+      if (e.key === 'Escape') {
+        onEscape();
+        e.preventDefault();
+      }
+      if ((e.metaKey || e.ctrlKey) && e.key === 'f') {
```

Comment:
In Connect we have customizable shortcuts, I think it should be customizable too.

This could be implemented by adding a callback `isSearchKeyboardEvent: (e: KeyboardEvent) => boolean` to this component. Web UI would pass `(e.metaKey || e.ctrlKey) && e.key === 'f'` and Connect would have another entry  in `appConfigSchema.ts` (like `keymap.terminalSearch`) and then you could check it through `keybaordShorcutsService.getShortcutAction(e) === 'terminalSearch'`.